TLM/TLM/UI/SubTools/LaneConnectorTool.cs

```diff
@@ -223,27 +295,29 @@ bool drawMarker
                                   || (laneMarker == selectedMarker)));
 
                     // highlight hovered marker and selected marker
-                    bool highlightMarker =
-                        drawMarker && ((laneMarker == selectedMarker) || markerIsHovered);
-
                     if (drawMarker) {
-                        laneMarker.Radius = highlightMarker ? 2f : 1f;
-                    } else {
-                        markerIsHovered = false;
-                    }
-
-                    if (markerIsHovered) {
-                        hoveredMarker = laneMarker;
-                    }
-
-                    var circleColor = laneMarker.IsTarget ? Color.white : laneMarker.Color;
+                        float hitH = TrafficManagerTool.GetAccurateHitHeight();
```

Comment:
This part of the code was a mess. so I cleaned it up a bit. the logic is the same.
The main differences are:
- I intersect mouse ray with segment lane marker.
- in the old code `IsLaneMarkerHovered()` was calculated more times than necessary. Now I only call it if I know I need it.